src/components/Chart.jsx

```diff
@@ -145,6 +146,90 @@ class Chart extends Component {
     }
 }
 
+Chart.propTypes = {
+    requestAPI                      : PropTypes.func.isRequired,
+    requestSubscribe                : PropTypes.func.isRequired,
+    requestForget                   : PropTypes.func.isRequired,
+    id                              : PropTypes.string,
+    granularity                     : PropTypes.oneOf([
+        0, 60, 120, 180, 300, 600,
+        900, 1800, 3600, 7200,
+        14400, 28800, 86400,
+    ]),
+    symbol                          : PropTypes.string,
+    chartType                       : PropTypes.oneOf(['mountain', 'line',
+        'colored_line', 'spline', 'baseline',
+        'candle', 'colored_bar', 'hollow_candle',
+        'heikinashi', 'kagi', 'linebreak',
+        'renko', 'rangebars', 'pandf']),
+    startEpoch                      : PropTypes.number,
+    endEpoch                        : PropTypes.number,
+    chartControlsWidgets            : PropTypes.any,
+    onSettingsChange                : PropTypes.func,
+    onMessage                       : PropTypes.func,
+    isMobile                        : PropTypes.bool,
+    enableRouting                   : PropTypes.bool,
+    removeAllComparisons            : PropTypes.bool,
+    topWidgets                      : PropTypes.func,
+    settings                        : PropTypes.object,
+    bottomWidgets                   : PropTypes.any,
+    DrawToolsSettingsDialog         : PropTypes.any.isRequired,
+    StudySettingsDialog             : PropTypes.any.isRequired,
+    isOnPagination                  : PropTypes.bool,
+    isChartAvailable                : PropTypes.bool,
+    barriers                        : PropTypes.array,
+    AggregateChartSettingsDialog    : PropTypes.any.isRequired,
+    chartContainerHeight            : PropTypes.number,
+    containerWidth                  : PropTypes.number,
+    isDrawing                       : PropTypes.bool,
+    theme                           : PropTypes.string,
+    position                        : PropTypes.string,
+    showLastDigitStats              : PropTypes.bool,
+    isConnectionOpened              : PropTypes.bool,
+    isAnimationEnabled              : PropTypes.bool,
+    scrollToEpoch                   : PropTypes.number,
+    scrollToEpochOffset             : PropTypes.number,
+    zoom                            : PropTypes.number,
+    clearChart                      : PropTypes.bool,
+    onExportLayout                  : PropTypes.func,
+    importedLayout                  : PropTypes.func,
+};
+
+Chart.defaultProps = {
+    id                              : undefined,
```

Comment:
@shayanbinary it doesn't need to set a default value when is `null`, `undefined`, `''` and `false`